tools/doc-generator/parse/parser.go

```diff
@@ -115,8 +115,12 @@ func Flags(cfg flagext.Registerer, logger log.Logger) map[uintptr]*flag.Flag {
 			return
 		}
 
-		ptr := reflect.ValueOf(f.Value).Pointer()
-		flags[ptr] = f
+		val := reflect.ValueOf(f.Value)
+		if val.CanAddr() {
+			flags[val.UnsafeAddr()] = f
+		} else if val.Kind() == reflect.Ptr {
+			flags[val.Pointer()] = f
+		}
```

Comment:
Should we log or handle the case where a value is neither addressable nor a pointe (even this unlikely to happen considering the current flags usage).

Reply:
Added one more option and a panic if we can't get the address